fmriprep/reports/core.py

```diff
@@ -22,100 +22,49 @@
-
-        if self.subject_id is not None:
-            self.root = self.root / f"sub-{self.subject_id}"
-
-        if "template_path" in settings:
-            self.template_path = config.parent / settings["template_path"]
-
-        self.index(settings["sections"])
-
-
-#
-# The following are the interface used directly by fMRIPrep
-#
-
-
-def run_reports(
-    out_dir,
-    subject_label,
-    run_uuid,
-    config=None,
-    reportlets_dir=None,
-    packagename=None,
-):
-    """
-    Run the reports.
-
-    .. testsetup::
-
-    >>> copytree_or_skip("data/tests/work", testdir)
-    >>> (testdir / 'fmriprep').mkdir(parents=True, exist_ok=True)
-
-    .. doctest::
-
-    >>> run_reports(testdir / 'out', '01', 'madeoutuuid', packagename='fmriprep',
-    ...             reportlets_dir=testdir / 'work' / 'reportlets' / 'fmriprep')
-    0
-
-    """
-    return Report(
-        out_dir,
-        run_uuid,
-        config=config,
-        subject_id=subject_label,
-        packagename=packagename,
-        reportlets_dir=reportlets_dir,
-    ).generate_report()
-
-
-def generate_reports(
-    subject_list, output_dir, run_uuid, config=None, work_dir=None, packagename=None
-):
-    """Execute run_reports on a list of subjects."""
+def generate_reports(subject_list, output_dir, run_uuid, config=None, work_dir=None):
+    """Generate reports for a list of subjects."""
     reportlets_dir = None
     if work_dir is not None:
         reportlets_dir = Path(work_dir) / "reportlets"
-    report_errors = [
-        run_reports(
+
+    errors = []
+    for subject_label in subject_list:
+        entities = {}
+        entities["subject"] = subject_label
+
+        robj = Report(
             output_dir,
-            subject_label,
             run_uuid,
-            config=config,
-            packagename=packagename,
+            bootstrap_file=config,
             reportlets_dir=reportlets_dir,
+            plugins=None,
+            plugin_meta=None,
+            metadata=None,
+            **entities,
         )
-        for subject_label in subject_list
-    ]
 
-    errno = sum(report_errors)
-    if errno:
-        import logging
-
-        logger = logging.getLogger("cli")
-        error_list = ", ".join(
-            f"{subid} ({err})" for subid, err in zip(subject_list, report_errors) if err
-        )
-        logger.error(
-            "Preprocessing did not finish successfully. Errors occurred while processing "
-            "data from participants: %s. Check the HTML reports for details.",
-            error_list,
+        # Count nbr of subject for which report generation failed
+        try:
+            robj.generate_report()
+        except:
+            import sys
+            import traceback
+
+            errors.append(subject_label)
+            traceback.print_exception(
+                *sys.exc_info(),
+                file=str(Path(output_dir) / "logs" / f"report-{run_uuid}-{subject_label}.err"),
+            )
+
```

Comment:
Apologies, I made the suggestion in the wrong place.
```suggestion
    return errors
```